python-client/giskard/core/model_validation.py

```diff
@@ -1,19 +1,24 @@
+import tempfile
 from typing import List, Iterable
 
+import yaml
 import numpy as np
 import pandas as pd
 from pandas.core.dtypes.common import is_string_dtype
 
+from giskard.core.core import ModelMeta
 from giskard.client.python_utils import warning
 from giskard.core.core import SupportedModelTypes
 from giskard.core.model import Model, WrapperModel
 from giskard.core.validation import validate_is_pandasdataframe, validate_target
 from giskard.ml_worker.core.dataset import Dataset
 
-
 def validate_model(model: Model, validate_ds: Dataset):
     model_type = model.meta.model_type
 
+    model_path = validate_model_saving(model)
+    loaded_model = validate_model_loading(model_path)
```

Comment:
Currently, if i set `model = validate_model_loading(model_path)`, i.e. if I continue the validation with the loaded model, I get the following error:
```py
python tests/tensorflow/test_tabular_titanic_binary_classification.py
2023-03-03 11:14:08.490405: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/.venv/lib/python3.7/site-packages/huggingface_hub/file_download.py:632: FutureWarning: `cached_download` is the legacy way to download files from the HF hub, please consider upgrading to `hf_hub_download`
  FutureWarning,
2023-03-03 11:14:16.950096: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
WARNING:tensorflow:Error in loading the saved optimizer state. As a result, your model is starting with a freshly initialized optimizer.
2023-03-03 11:14:17,019 pid:61814 MainThread tensorflow   WARNING  Error in loading the saved optimizer state. As a result, your model is starting with a freshly initialized optimizer.
2023/03/03 11:14:17 WARNING mlflow.tensorflow: You are saving a TensorFlow Core model or Keras model without a signature. Inference with mlflow.pyfunc.spark_udf() will not work unless the model's pyfunc representation accepts pandas DataFrames as inference inputs.
INFO:tensorflow:Assets written to: /var/folders/jp/b7681vg128nf8s2hw47sl6380000gn/T/giskard-model-srw1gzrp/data/model/assets
2023-03-03 11:14:17,514 pid:61814 MainThread tensorflow   INFO     Assets written to: /var/folders/jp/b7681vg128nf8s2hw47sl6380000gn/T/giskard-model-srw1gzrp/data/model/assets
/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/.venv/lib/python3.7/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
Traceback (most recent call last):
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/giskard/core/model_validation.py", line 56, in validate_model_execution
    prediction = model.predict(validation_ds)
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/giskard/core/model.py", line 162, in predict
    raw_prediction = self.predict_df(df)
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/giskard/core/model.py", line 329, in predict_df
    raw_prediction = self.clf_predict(df)
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/giskard/models/tensorflow/__init__.py", line 41, in clf_predict
    return self.clf.predict(data)
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/.venv/lib/python3.7/site-packages/keras/utils/traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/.venv/lib/python3.7/site-packages/tensorflow/python/framework/constant_op.py", line 102, in convert_to_eager_tensor
    return ops.EagerTensor(value, ctx.device_name, dtype)
ValueError: Failed to convert a NumPy array to a Tensor (Unsupported object type int).

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "tests/tensorflow/test_tabular_titanic_binary_classification.py", line 56, in <module>
    test_tabular_titanic_binary_classification()
  File "tests/tensorflow/test_tabular_titanic_binary_classification.py", line 52, in test_tabular_titanic_binary_classification
    tests.utils.verify_model_upload(my_model, my_test_dataset)
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/tests/utils.py", line 48, in verify_model_upload
    my_model.upload(client, "test-project", my_data)
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/giskard/core/model.py", line 205, in upload
    validate_model(model=self, validate_ds=validate_ds)
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/giskard/core/model_validation.py", line 49, in validate_model
    validate_model_execution(model, validate_ds)
  File "/Users/rak/Documents/giskard_repos/giskard_huggingface/python-client/giskard/core/model_validation.py", line 62, in validate_model_execution
    ) from e
ValueError: Invalid model.predict() input.
Please make sure that model.predict(dataset) does not return an error message before uploading in Giskard

```

Reply:
This is fixed now.